client/my-sites/domains/components/domain-warnings/pending-gsuite-tos-notice-dialog.jsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+import PropTypes from 'prop-types';
+import React, { Fragment, useState } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import Button from 'components/button';
+import ClipboardButton from 'components/forms/clipboard-button';
+import { composeAnalytics, recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+import Dialog from 'components/dialog';
+import { getLoginUrlWithTOSRedirect } from 'lib/google-apps';
+import VerticalNav from 'components/vertical-nav';
+import VerticalNavItem from 'components/vertical-nav/item';
+import wp from 'lib/wp';
+
+function PendingGSuiteTosNoticeDialog( props ) {
+	const [ password, setPassword ] = useState( false );
+	const [ isCopied, setIsCopied ] = useState( false );
+	const translate = useTranslate();
+
+	const onPasswordClickHandler = e => {
+		e.preventDefault;
+		const wpcom = wp.undocumented();
+		const mailbox = props.user.split( '@' )[ 0 ];
+		wpcom.resetPasswordForMailbox( props.domainName, mailbox ).then( data => {
+			setPassword( data.password );
+		} );
+	};
+
+	const recordLogInClick = () => {
+		props.pendingAccountLogInClick( {
+			domainName: props.domainName,
+			isMultipleDomains: props.isMultipleDomains,
+			user: props.user,
+			severity: props.severity,
+			section: props.section,
+			siteSlug: props.siteSlug,
+		} );
+	};
+
+	const onCopyAction = () => {
+		setIsCopied( true );
+	};
+
+	return (
+		<Dialog className="domain-warnings__dialog" isVisible={ props.visible }>
+			<header>
+				<h1>{ translate( 'Finish G Suite Setup' ) }</h1>
+				<button onClick={ props.onClose }>
+					<Gridicon icon="cross" />
+				</button>
+			</header>
+
+			<p>
```

Comment:
With these big conditional blocks, it feels like we should move them into their own render functions.

```
{ password ? this.renderHasPassword() : this.renderResetPassword() }
```